src/PIL/TgaImagePlugin.py

```diff
@@ -118,7 +118,7 @@ def _open(self) -> None:
             start, size, mapdepth = i16(s, 3), i16(s, 5), s[7]
             if mapdepth == 16:
                 self.palette = ImagePalette.raw(
-                    "BGR;15", b"\0" * 2 * start + self.fp.read(2 * size)
+                    "BGRA;15", b"\0" * 2 * start + self.fp.read(2 * size)
                 )
```

Comment:
```suggestion
                )
                self.palette.mode = "RGBA"
```
BGRA;15 can be unpacked, but only to RGBA
https://github.com/python-pillow/Pillow/blob/489795422381cd6040aa1efa4e112c9c2975fa82/src/libImaging/Unpack.c#L1620

The change I've suggested here, along with https://github.com/python-pillow/Pillow/pull/7966, will allow the test suite to start passing.